File: BOJ/5525/A.py
import sys

sys.stdin = open(
    '/Users/sungwonlee/Documents/Projects/algorithms-practice/BOJ/5525/input.txt', 'r')

N = int(input())
M = int(input())
S = input()
# print(N)
# print(M)
# print(S)
T = ''
for i in range(0, N):
    T = T + 'IO'
T = T + 'I'
C = ''
cnt = 0
idx = 0
while(True):
    idx = S.find(T, idx)
    if(idx == -1):
        break
    cnt = cnt + 1
    next_idx = idx+len(T)
    while(next_idx <= len(S)-2 and S[next_idx:next_idx+2] == "OI"):
        cnt = cnt+1
        next_idx = next_idx + 2
    idx = next_idx


print(cnt)

File: BOJ/5525/A.cpp
#include <iostream>
#include <string>
using namespace std;

int main()
{
    freopen("input.txt", "rt", stdin);
    ios::sync_with_stdio(0);
    cin.tie(0);
    int N;
    cin >> N;
    int L;
    cin >> L;
    string S;
    cin >> S;
    string to_find = "I";
    for (int i = 0; i < N; i++)
    {
        to_find += "OI";
    }
    int f_len = to_find.size();
    string s = S;
    int cnt = 0;
    int start = 0;
    // initial starting point
    int idx = s.find(to_find);
    while (idx != -1)
    {
        cnt++;
        int i = idx + f_len;
        // check for more
        while (i < s.size() - 1 && string(1, s[i]) + string(1, s[i + 1]) == "OI")
        {
            i += 2;
            cnt++;
        }
        // if i is out of bounds, break
        if (i >= L - 1)
        {
            break;
        }
        idx = s.find(to_find, i);
    }
    cout << cnt;
}